source/async-return-type.d.ts

```diff
@@ -0,0 +1,19 @@
+import {PromiseValue} from './promise-value';
+
+type AsyncFunction = (...args: unknown[]) => Promise<unknown>;
+
+/**
+Unpack the return type of a function that returns a `Promise`.
+
+@example
+```ts
+import {AsyncReturnType} from 'type-fest';
+import {asyncFunction} from 'api';
+
+// This type resolves to the unpacked return type of `asyncFunction`.
+type Value = AsyncReturnType<typeof asyncFunction>;
+
+let value: Value = await asyncFunction();
```

Comment:
> Example code block with a realistic example.

I wouldn't say this is a very realistic example. Look at some of the other types to see how to do it.